### PR TITLE
feat(waf): add a datasource to get the global protection whitelist rule list

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a datasource to get the global protection whitelist rule list.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of the global protection whitelist rules.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/waf" TESTARGS="-run TestAccDataSourceRulesGlobalProtectionWhitelist_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDataSourceRulesGlobalProtectionWhitelist_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceRulesGlobalProtectionWhitelist_basic
=== PAUSE TestAccDataSourceRulesGlobalProtectionWhitelist_basic
=== CONT  TestAccDataSourceRulesGlobalProtectionWhitelist_basic
--- PASS: TestAccDataSourceRulesGlobalProtectionWhitelist_basic (440.40s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       440.495s
```

### CODE REVIEW

File: docs/data-sources/waf_rules_global_protection_whitelist.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+* `policy_id` - (Required, String) Specifies the ID of the policy to which the global protection whitelist rules belong.
+
+* `rule_id` - (Optional, String) Specifies the ID of the global protection whitelist rule.
+
+* `status` - (Optional, String) Specifies the status of the global protection whitelist rule.
+  The valid values are as follows:
+  + **0**: The global protection whitelist rule is disabled.
+  + **1**: The global protection whitelist rule is active.
+
+* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID to which the protection policies belong.
```
Comment (jinyangyang222) [inline docs/data-sources/waf_rules_global_protection_whitelist.md:41]:
```suggestion
* `enterprise_project_id` - (Optional, String) Specifies the enterprise project ID to which the protection policy belong.
```

Comment (ruwenqiang123) [reply docs/data-sources/waf_rules_global_protection_whitelist.md]: done

**Pull request merged**